crates/tako/src/server/monitoring.rs

```diff
@@ -0,0 +1,73 @@
+use crate::messages::gateway::LostWorkerReason;
+use crate::messages::worker::WorkerOverview;
+use crate::server::core::CoreRef;
+use crate::WorkerId;
+
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct MonitoringEvent {
+    pub message: MonitoringEventsPayload,
+    pub event_id: u32,
+    pub event_time: SystemTime,
```

Comment:
```suggestion
    id: u32,
    time: SystemTime,
```

`id` and `time` should not be publicly writable. There should be some function on an event manager that will just receive the payload, and the manager will automatically assign the time and ID.